src/main/java/seedu/address/MainApp.java

```diff
@@ -76,7 +78,10 @@ public void init() throws Exception {
     private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
         Optional<ReadOnlyAddressBook> addressBookOptional;
         ReadOnlyAddressBook initialData;
+
         try {
+            Block.setBlockPref(userPrefs.getBlockSettings());
+            Room.setRoomPref(userPrefs.getRoomSettings());
```

Comment:
Should `setBlockPref` and `setRoomPref` be handled in `ModelManager` to follow the separation of concerns principle? As `ModelManager` handles `Model` related functions.

Reply:
It needs to be set before the `addressBook` is loaded else there will be a validation error when loading the saved `addressBook`

Reply:
Okay, I see what you are coming from. As `storage.readAddressBook()` reads `addressbook.json` and converts `JsonAdapterPerson` to `Person`, it will do the validation check for every `Person` field.
Do remember to update `JsonAdapterPerson#toModelType()` to do a validation check for `Block` and `Room` before declaring it.

Reply:
`userPref` will never be `null`, because the check is done in `UserPref` . `Block` and `Room` in this case is referring to the class, so it cant be `null` either

Reply:
Oh, I refer to `JsonAdapterPerson#toModelType()` method, the current convention will do a null and validation check for every field when converting `JsonAdapterPerson` to `Person`.
There is missing of those checks for `Block` and `Room`.

Reply:
Oh, I refer to `JsonAdaptedPerson#toModelType` method. The current convention will do a null and validation check for every field before converting the `JsonAdapterPerson` to `Person`.
There is missing of null and validation check for `block` and `room` String.